src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -193,24 +204,36 @@ protected long getTargetSSRC()
      * @return true if the stream represented by incomingFrameBaseLayerIndex
      * represents a step closer (or all the way) to the targetBaseLayerIndex
      */
-    private static boolean shouldSwitchToBaseLayer(
+    private boolean shouldSwitchToBaseLayer(
+        long nowMs,
         int currentBaseLayerIndex,
         int incomingFrameBaseLayerIndex,
         int targetBaseLayerIndex)
     {
-        if ((currentBaseLayerIndex < incomingFrameBaseLayerIndex) &&
-            incomingFrameBaseLayerIndex <= targetBaseLayerIndex)
+        if (nowMs - lastSwitch > MIN_KEY_FRAME_WAIT_MS)
         {
-            // upscale case
+            lastSwitch = nowMs;
             return true;
         }
-        else if ((currentBaseLayerIndex > incomingFrameBaseLayerIndex) &&
-            (incomingFrameBaseLayerIndex >= targetBaseLayerIndex))
+        else
         {
-            // downscale case
-            return true;
+            // Upscale/downscale happens in a window of 300ms since the
+            // reception of the first key frame of a key frame group.
```

Comment:
this comment seems incomplete?

Reply:
I reworded the comment, let me know if it needs more attention.